core/src/main/scala/com/databricks/labs/remorph/graph/TableGraph.scala

```diff
@@ -0,0 +1,187 @@
+package com.databricks.labs.remorph.graph
+
+import com.databricks.labs.remorph.discovery.{ExecutedQuery, QueryHistory, TableDefinition}
+import com.databricks.labs.remorph.parsers.PlanParser
+import com.databricks.labs.remorph.transpilers.{Result, SourceCode}
+import com.typesafe.scalalogging.LazyLogging
+import com.databricks.labs.remorph.{intermediate => ir}
+
+protected case class Node(tableDefinition: TableDefinition, metadata: Map[String, Set[String]])
+// `from` is the table which is sourced to create `to` table
+protected case class Edge(from: Node, to: Option[Node], metadata: Map[String, String])
+
+class TableGraph(parser: PlanParser[_]) extends DependencyGraph with LazyLogging {
+  private val nodes = scala.collection.mutable.Set.empty[Node]
+  private val edges = scala.collection.mutable.Set.empty[Edge]
+
+  override protected def addNode(id: TableDefinition, metadata: Map[String, Set[String]]): Unit = {
+    // Metadata list of query ids and add node only if it is not already present.
```

Comment:
can we avoid free-form string-to-string maps and add explicit `queryIds: Set[String]`?